File: packages/scripts/src/verification/verify-game.ts
import { chromium, type Page } from 'playwright';
import * as path from 'node:path';

async function testGameLoad(page: Page) {
    console.log("Navigating to game...");
    // Assuming local dev server port, should be configurable
    await page.goto("http://localhost:4323/neo-tokyo-rival-academies");

    // Wait for the canvas element which confirms 3D scene load
    await page.waitForSelector("canvas", { timeout: 30000 });

    console.log("Canvas found. Waiting for splash screen (5s)...");
    await page.waitForTimeout(5000);

    await page.screenshot({ path: path.join(process.cwd(), "verification/game_menu.png") });
    console.log("Menu screenshot taken.");

    // Look for START STORY button
    try {
        const startBtn = page.getByText("INITIATE STORY MODE");
        if (await startBtn.isVisible()) {
            await startBtn.click();
            console.log("Clicked INITIATE STORY MODE");

            // Wait for Intro Overlay
            await page.waitForTimeout(2000);
            await page.screenshot({ path: path.join(process.cwd(), "verification/game_intro.png") });

            // Click to advance intro
            console.log("Advancing intro...");
            for (let i = 0; i < 10; i++) {
                await page.mouse.click(640, 360);
                await page.waitForTimeout(300);
            }

            console.log("Intro should be done. Waiting for gameplay...");
            await page.waitForTimeout(3000);
            await page.screenshot({ path: path.join(process.cwd(), "verification/game_gameplay.png") });
            console.log("Gameplay screenshot taken.");
        } else {
            console.log("Start button not visible.");
        }
    } catch (e) {
        console.error(`Error: ${e}`);
    }
}

async function main() {
    const browser = await chromium.launch({ headless: true });
    const context = await browser.newContext({ viewport: { width: 1280, height: 720 } });
    const page = await context.newPage();

    try {
        // Ensure verification dir exists
        // fs.mkdirSync ... (omitted for brevity, running in env likely has it or we can add)
        await testGameLoad(page);
    } finally {
        await browser.close();
    }
}

main().catch(console.error);

File: verify_game.py
import asyncio
from playwright.async_api import async_playwright
import os

async def verify_game():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        page = await browser.new_page()

        # Start server (assuming it's running on 4321)
        try:
            await page.goto("http://localhost:4321/neo-tokyo-rival-academies", timeout=20000)
            print("Successfully loaded http://localhost:4321")

            # Wait for canvas
            await page.wait_for_selector("canvas", timeout=20000)
            print("Canvas element found")

            # Check for Main Menu text (after Splash)
            # Wait for main menu text to appear after splash
            try:
                await page.wait_for_selector("text=NEO-TOKYO", timeout=10000)
                print("Main Menu text found")
            except Exception:
                print("WARNING: Main Menu text NOT found within timeout")

            # Take screenshot
            os.makedirs("screenshots", exist_ok=True)
            await page.screenshot(path="screenshots/game_screen.png")
            print("Screenshot saved to screenshots/game_screen.png")

        os.makedirs("screenshots", exist_ok=True)
        try:
            await page.goto("http://localhost:4321/neo-tokyo-rival-academies", timeout=20000)
            print("Successfully loaded http://localhost:4321/neo-tokyo-rival-academies")

            # Wait for canvas
            await page.wait_for_selector("canvas", timeout=20000)
            print("Canvas element found")

            # Check for Main Menu text (after Splash)
            # Splash takes ~3s.
            await asyncio.sleep(4)
            content = await page.content()
            if "NEO-TOKYO" in content:
                print("Main Menu text found")
            else:
                print("WARNING: Main Menu text NOT found")

            # Take screenshot
            await page.screenshot(path="screenshots/game_screen.png")
            print("Screenshot saved to screenshots/game_screen.png")

        except (TimeoutError, Exception) as e:
            print(f"Error: {e}")
            # Take screenshot of error
            await page.screenshot(path="screenshots/error_screen.png")

        finally:
            await browser.close()

if __name__ == "__main__":
    asyncio.run(verify_game())
